### PR TITLE
Feature/us16548 analytics

### PR DESCRIPTION
**Pull request by phritolay**

### CODE REVIEW

File: src/app/sign-up-for-project/sign-up-for-project.component.html [hunk 1]
```diff
@@ -80,8 +80,7 @@ <h5>{{project.spots}}<sup>*</sup></h5>
         </div>
       </div>
 
-           
-      <form [formGroup]="signUpForm" action="/" (ngSubmit)="onSignUpSubmission();" novalidate>
+      <form [formGroup]="signUpForm" action="/" novalidate>
```
Comment (vchtelmakh) [inline src/app/sign-up-for-project/sign-up-for-project.component.html]: Good catch!

File: src/app/sign-up-for-project/sign-up-for-project.component.ts [hunk 2]
```diff
@@ -73,11 +75,28 @@ export class SignUpForProjectComponent implements OnInit {
   }
 
   onSignUpSubmission(): void {
-    this.isSubmitted = true;
+    const guest = this.signUpForm.value.guests;
```
Comment (vchtelmakh) [inline src/app/sign-up-for-project/sign-up-for-project.component.ts]: `const guest` --> `const guests`

File: src/app/sign-up-for-project/sign-up-for-project.component.ts [hunk 3]
```diff
@@ -73,11 +75,28 @@ export class SignUpForProjectComponent implements OnInit {
...
+    const guestHouseholdPosition = []
 
-    this.signUpForm.valid ?
-      this.saveVolunteerSignUp(this.signUpForm, this.projectId, this.countOfKidsUnderEight, this.countOfKidsUnderThirteen) :
-      this.signUpForm = this.formHelper.markAllFieldsAsTouched(this.signUpForm);
+    guest.forEach(function (guest) {
```
Comment (vchtelmakh) [inline src/app/sign-up-for-project/sign-up-for-project.component.ts]: Please use arrow functions `guests.forEach(guest => {...`

File: src/app/sign-up-for-project/sign-up-for-project.component.ts [hunk 4]
```diff
@@ -73,11 +75,28 @@ export class SignUpForProjectComponent implements OnInit {
...
+    this.analyticsService.trackProjectSubmitted({
+      Id: this.project.id,
+      KidsUnderEight: this.countOfKidsUnderEight,
+      KidsUnderThirteen: this.countOfKidsUnderThirteen,
+      VolunteersThirteenPlus: this.signUpForm.controls.guests.value.length,
+      GuestsHouseholdPositions: guestHouseholdPosition
+    });
+
+    setTimeout(() => {
+      this.isSubmitted = true;
```
Comment (vchtelmakh) [inline src/app/sign-up-for-project/sign-up-for-project.component.ts]: This shows errors and aborts the submission if the form is invalid. I don't think we should set it after a second. Also, I am not sure if we should send analytics events if the submission is an attempt with invalid fields.

Comment (phritolay) [reply src/app/sign-up-for-project/sign-up-for-project.component.ts]: 👍 👍  We pulled the `setTimeout` from another form on the site and were looking for a better way to accomplish this.

**Pull request merged**